.github/workflows/pre-commit.yml

```
@@ -7,6 +7,10 @@ on:
   pull_request:
     paths-ignore:
       - '**.md'
+concurrency:
+  group: pre-commit-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Do we want to cancel jobs when running on `main`? I'd say every commit in `main` must be tested. If there's a regression, we want to know where it started.

---

@kvaps, you can leave this to a separate issue, which I'll take later, if you don't mind.

---

same with release branches, we should run tests on them

It can be something like this, though I haven't tested it yet

```suggestion
  group: ${{ (github.ref == 'refs/heads/main' || startsWith (github.ref, 'refs/heads/release') }} && 
      pre-commit-${{ github.run_id }}-${{ github.run_attempt }} ||
      pre-commit-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

---

```suggestion
  group: ${{
    (github.ref == 'refs/heads/main' || startsWith (github.ref, 'refs/heads/release') &&
    format('pre-commit-{0}-{1}, github.run_id, github.run_attempt) ||
    format('pre-commit-{0}-{1}, github.workflow, github.event.pull_request.number || github.ref)
  }}
```

---

I'd disable pre-commit checks on release branches, we will run tests during the release cylce any way